app/main/views/frameworks.py

```diff
@@ -109,6 +132,53 @@ def update_framework(framework_slug):
             enforce_required=True
         )
 
+    if json_payload['frameworks'].get('status', None) == 'pending' and framework.status != 'pending':
+        # When we set a framework to pending, we want to polyfill all supplier declarations (regardless of whether
+        # their application is 'complete' or not) with information from their supplier account. We have a set of
+        # hardcoded keys for these values. Currently these keys don't need to mean anything outside of the API, so
+        # they are defined in dm-api/app/constants. At the time this was written, it didn't seem feasible to manage
+        # this query through SQLAlchemy in an efficient (i.e. 100% server-side) manner, so we are using raw SQL.
+
+        # We pump this data into the declaration before updating the framework state, so that if for some reason the
+        # framework state fails to update, when a second attempt is made the data is overwritten.
+        statement = text(f"""
+UPDATE
+    supplier_frameworks sf
+SET
+    declaration = sf.declaration::jsonb || (
+        SELECT
+            row_to_json(data)
+        FROM
+        (
+            SELECT
+                s.name AS "{KEY_TRADING_NAME}",
+                s.registered_name AS "{KEY_REGISTERED_NAME}",
+                COALESCE(s.companies_house_number,
+                         s.other_company_registration_number) AS "{KEY_REGISTRATION_NUMBER}",
+                s.registration_date AS "{KEY_REGISTRATION_DATE}",
+                s.duns_number AS "{KEY_DUNS_NUMBER}",
+                s.vat_number AS "{KEY_VAT_NUMBER}",
+                s.trading_status AS "{KEY_TRADING_STATUS}",
+                s.organisation_size AS "{KEY_ORGANISATION_SIZE}",
+                s.registration_country AS "{KEY_REGISTRATION_COUNTRY}",
+                ci.address1 AS "{KEY_REGISTRATION_BUILDING}",
+                ci.city AS "{KEY_REGISTRATION_TOWN}",
+                ci.postcode AS "{KEY_REGISTRATION_POSTCODE}"
+            FROM
+                suppliers AS s
```

Comment:
Is there a reason for abbreviating stuff in this query?

Reply:
Not really, it's just a shorthand I've become accustomed to when writing SQL. I could get rid of it if you find it hard to follow?